java/connectors/semantickernel-connectors-memory-azurecognitivesearch/src/main/java/com/microsoft/semantickernel/connectors/memory/azurecognitivesearch/AzureCognitiveSearchMemory.java

```diff
@@ -323,8 +326,7 @@ private static String normalizeNullToEmptyString(@Nullable String value) {
     // <returns>Normalized name</returns>
     private static String normalizeIndexName(String indexName) {
         if (indexName.length() > 128) {
-            throw new AzureCognitiveSearchMemoryException(
-                    "The collection name is too long, it cannot exceed 128 chars");
+            throw new IllegalArgumentException("The collection name cannot exceed 128 chars");
```

Comment:
Seems to have gone less specific

Reply:
One of the points in the error handling ADR is about unnecessary exceptions, the concern being that clients may become dependent on them making it harder to change in the future. I felt that the AzureCognitiveSearchMemoryException was such a case as it was only used in this case, and two others which could be up-leveled. Further, I felt that this was not truly a "memory exception" as this is not something unexpected that happened during processing, such as not being able to parse a JSON value. This is a case where the programmer has done the wrong thing and, as such, this should be caught during development. Still, the javadoc does not reflect this limitation so I will fix the javadoc for this.